packages/google-drive-kit/src/board-server/operations.ts

```diff
@@ -179,18 +184,59 @@ class DriveOperations {
     const api = new Files({ kind: "key", key: apiKey });
     const fileRequest = await retryableFetch(api.makeQueryRequest(query));
     const response = (await fileRequest.json()) as DriveFileQuery;
-    const results = await Promise.all(
-      response.files.map(async (file) => {
-        const properties = readProperties(file);
-        const tags: GraphTag[] = ["featured"];
-        const thumbnailUrl = properties.thumbnailUrl;
-        const title = file.name.replace(/(\.bgl)?\.json$/, "");
-        return { id: file.id, title, tags, thumbnailUrl };
-      })
+
+    const results = (await this.toGraphInfos(api, response.files)).map(
+      (graphInfo) => {
+        if (!graphInfo.tags.includes("featured")) {
+          // The fact that a graph is in the featured folder alone determines
+          // whether it is featured.
+          graphInfo.tags.push("featured");
+        }
+        return graphInfo;
+      }
     );
     return results;
   }
 
+  async toGraphInfos(
+    api: Files,
+    files: Array<DriveFile>
+  ): Promise<Array<GraphInfo>> {
+    const maybeFetchThumbnail = async (file: DriveFile) => {
+      const appProperties = readAppProperties(file);
+      if (file.properties?.thumbnailUrl) {
+        return { file, appProperties, thumbnail: file.properties!.thumbnailUrl };
+      }
+      const thumbnailUrl = appProperties.thumbnailUrl;
+      if (thumbnailUrl) {
+        const thumbnailFileId = getFileId(thumbnailUrl);
+        const response = await retryableFetch(
```

Comment:
This basically rolls all the fetches into a single call that blocks loading. Ok for now, but I think we will need to figure out how to make this go asynchronously, especially when people have a large number of boards.

Reply:
Yes, I have a way in mind - return here and fetch from the UI layer after the list is rendered showing generic icons in meanwhile. This is how it's done in the theme generator currently.